client/packages/core/src/instatx.ts

```diff
@@ -15,7 +20,18 @@ export interface TransactionChunk {
    *  const goalId = id();
    *  tx.goals[goalId].update({title: "Get fit", difficulty: 5})
    */
-  update: (args: { [attribute: string]: any }) => TransactionChunk;
+  update: (
+    args: {
+      [K in keyof S["entities"][E]["attrs"]]?: S["entities"][E]["attrs"][K] extends DataAttrDef<
```

Comment:
nit: would abstract the types for `update` `link` and `unlink`, to as they may look a bit unwieldy in intellisense. 

One stackoverflow that may be of interest to you: 

https://stackoverflow.com/questions/77867172/is-it-possible-to-produce-outputt-which-visually-omits-some-keys-in-t